zeebe/broker/src/test/java/io/camunda/zeebe/broker/partitioning/topology/ClusterChangeExecutorImplTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.broker.partitioning.topology;
+
+import io.camunda.zeebe.broker.exporter.repo.ExporterLoadException;
+import io.camunda.zeebe.broker.exporter.repo.ExporterRepository;
+import io.camunda.zeebe.exporter.api.Exporter;
+import io.camunda.zeebe.exporter.api.context.Context;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.scheduler.testing.TestConcurrencyControl;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.util.ArrayList;
+import java.util.List;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ClusterChangeExecutorImplTest {
+
+  @Test
+  public void shouldRunPurgeForEveryExporter() {
+    // given
+    final ExporterRepository repository = new ExporterRepository();
+    try {
+      repository.validateAndAddExporterDescriptor("test-1", AuditExporter.class, null);
+      repository.validateAndAddExporterDescriptor("test-2", AuditExporter.class, null);
+    } catch (final ExporterLoadException e) {
+      Assertions.fail(e);
+    }
+
+    // when
+    final var executor =
+        new ClusterChangeExecutorImpl(
+            new TestConcurrencyControl(), repository, new SimpleMeterRegistry());
+    executor.deleteHistory().join();
```

Comment:
:wrench: You can omit the join here, and just let AssertJ run its assertions on the future, e.g.

```java
    final var executor =
        new ClusterChangeExecutorImpl(
            new TestConcurrencyControl(), repository, new SimpleMeterRegistry());
    final Future<Void> result = executor.deleteHistory();

    // then
    Assertions.assertThat(result).succeedsWithin(Duration.ofSeconds(5));
    Assertions.assertThat(AuditExporter.AUDITS)
        .containsSubsequence("configure-test-1", "purge-test-1", "close-test-1");
    Assertions.assertThat(AuditExporter.AUDITS)
        .containsSubsequence("configure-test-2", "purge-test-2", "close-test-2");
```